category_encoders/target_encoder.py

```diff
@@ -113,34 +127,52 @@ def _fit(self, X, y, **kwargs):
 
     def fit_target_encoding(self, X, y):
         mapping = {}
+        prior = self._mean = y.mean()
 
         for switch in self.ordinal_encoder.category_mapping:
             col = switch.get('col')
-            values = switch.get('mapping')
+            if 'HEIR_' not in col:
+                values = switch.get('mapping')
 
-            prior = self._mean = y.mean()
+                scalar = prior
+                if self.heirarchy and col in self.heirarchy:
+                    col_heir = 'HEIR_'+col
+                    stats_heir = y.groupby(X[col_heir]).agg(['count', 'mean'])
+                    smoove_heir = self._weighting(stats_heir['count'])
+                    scalar_heir = scalar * (1 - smoove_heir) + stats_heir['mean'] * smoove_heir
+                    scalar_heir_long = X[[col, col_heir]].drop_duplicates()
+                    scalar_heir_long.index = np.arange(1, scalar_heir_long.shape[0]+1)
+                    scalar = scalar_heir_long[col_heir].map(scalar_heir.to_dict())
+                    X.drop([col_heir], axis=1, inplace=True)
+                    # Added line to reduce dimension of ordinal encoder - Whats the third dimension?
+                    self.ordinal_encoder._dim -= 1
+                    self.ordinal_encoder.cols.remove(col_heir)
 
-            stats = y.groupby(X[col]).agg(['count', 'mean'])
 
-            smoove = 1 / (1 + np.exp(-(stats['count'] - self.min_samples_leaf) / self.smoothing))
-            smoothing = prior * (1 - smoove) + stats['mean'] * smoove
-            smoothing[stats['count'] == 1] = prior
+                stats = y.groupby(X[col]).agg(['count', 'mean'])
+                smoove = self._weighting(stats['count'])
 
-            if self.handle_unknown == 'return_nan':
-                smoothing.loc[-1] = np.nan
-            elif self.handle_unknown == 'value':
-                smoothing.loc[-1] = prior
+                smoothing = scalar * (1 - smoove) + stats['mean'] * smoove
+                smoothing[stats['count'] == 1] = scalar
 
-            if self.handle_missing == 'return_nan':
-                smoothing.loc[values.loc[np.nan]] = np.nan
-            elif self.handle_missing == 'value':
-                smoothing.loc[-2] = prior
+                if self.handle_unknown == 'return_nan':
+                    smoothing.loc[-1] = np.nan
+                elif self.handle_unknown == 'value':
+                    smoothing.loc[-1] = prior
 
-            mapping[col] = smoothing
+                if self.handle_missing == 'return_nan':
+                    smoothing.loc[values.loc[np.nan]] = np.nan
+                elif self.handle_missing == 'value':
+                    smoothing.loc[-2] = prior
+
+                mapping[col] = smoothing
+
+        self.ordinal_encoder.mapping = [map for map in self.ordinal_encoder.mapping if not "HEIR_" in map['col']]
```

Comment:
This is basically cleaning the ordinal encoder so that it never new there was hierarchy involved? I like the idea that all the additional columns should be hidden from the user

Reply:
As above, this line has been removed in the new method.